src/opm/input/eclipse/EclipseState/Tables/TableManager.cpp

```diff
@@ -1562,7 +1562,7 @@ std::optional<JFunc> make_jfunc(const Deck& deck) {
                 try {
                     std::shared_ptr<TableType> table = std::make_shared<TableType>(dataItem, useJFunc(), tableIdx );
                     container.addTable( tableIdx , table );
-                } catch (const std::runtime_error& err) {
+                } catch (const std::invalid_argument& err) {
```

Comment:
I don't think this is a valid transformation.  You'll have to ensure that any exceptions thrown from the `TableType` constructor (or the `addTable()` function call, for that matter) that are **not** `invalid_argument` can safely propagate to the caller.  Unless you've already done that I'd be wary to do this.